### PR TITLE
HeistAction

### PR DESCRIPTION
**Pull request by BrianWasTkn**
- This is a...
  - [ ] Superaction    
  - [x] Action
  - [ ] Snippet
- [x] I have followed the Guidelines for Contributing as closely as possible.
- [x] I have thoroughly tested my (super)action and ensured that it works properly, even when used incorrectly.
- [x] My action is not a wrapper/alias for another command - it has an actual use.

--------------------------------------------------------------------------------
Just ensuring no one would try and steal or pirate the code ever again.

### CODE REVIEW

**Comment by JaM3141** [comment]
stalebot repellant

**Comment by JaM3141** [comment]
This PR seems to have a lot of additional changes to actions that are not related to the PR title. If you could outline exactly what those changes are so we can more effectively review those too, that would be great.

**Comment by BrianWasTaken** [comment]
updated my pr comment

**Review by JaM3141 — CHANGES_REQUESTED** [review]
- Your initial `{set}`s should probably have a `CHANGE_THIS_VALUE` type thing instead of a valid role ID, just so they know exactly what to edit.
- The helper `{set}`s are a tiny bit confusing. If you follow the Instructions, it shouldn't be too much of a problem but they're kinda grouped in with the `{set}`s you *are* supposed to edit.
- `a!heist help` doesn't display a help embed (Guidelines point 3). It thinks `help` is an invalid user.
- When you run `a!heist <user> <amount>`, it says `@user1 please the balance of @user2`, I have no clue what that means. I see there's a `{channel.awaitmessage}` check looking for `pls bal` something something. Might be intentional, might not be. But I am super confused by it. `pls explain` hahahaha get it
- The withdraw amount is set at a fixed 2,000. Is that intentional?

**Comment by BrianWasTaken** [comment]
done i guess? lol bye

**Comment by Shylke** [comment]
Hey Brian I ended up reviewing this;

* `Please the balance of @User` should be "Please check the balance of @User`

**Comment by Shylke** [comment]
Hey Brian I decided to review your action instead;

* `Please the balance of @User` should be `Please check the balance of @User`, or even `Please check @User's balance by running "pls bal @User"` to make it clear you're looking for `pls bal`/ Dank Memer in your code.
*  When not running `pls bal` Atlas says `did not checked user balance` "checked" should be "check"
* Withdrawal amount is fixed at 1000. I thought it was an error on my end for not having Dank Memer but inviting it and running 'pls bal @User` doesn't fix that (also because the balance command is an embed and I'm not so sure you can `{find}` in embeds)
* You can run `a!heist @Sponsor 10` and then then be able to check the balance of another user and have Atlas still have run the command. This is very minor and actually not that much of a problem since it doesn't impact the action's functioning, tbh I don't know why the pseudo Dank Memer integration exists at all
* Not setting role id will make the `{role.mention}` error out, might want to `{catch}` them


This is more of a me thing, but the delays before deleting messages are *way* too short, it made it harder for me to test this action as most error/warnings would be gone in the blink of an eye

**Comment by BrianWasTaken** [comment]
OOP, alright, fixing it, i'll comment again later on 'til i push out a fix.

**Comment by BrianWasTaken** [comment]
Here are my fixes for each of those you pointed out, respectively:

- I added a `{get}` for optional balance checking.
- Yes.
- Replaced the entire field with the custom message.
- Going back to point #1, there's an option to check balances.
- Catch tag added, wasn't happy before but now I'm liking it.

**Comment by DracoClaw** [comment]
Aight, finally reviewing this shit. Sorry. 

- Firstly, you need to make it VERY CLEAR that this action relies on another bot, ie Dank Memer. Please state this in the README as well as a `{note}` at the top of the action.
- Please remove the server specific IDs from the action, leave them black or have ROLE_ID
- Having the `Delete Invocation Message` flag on by default is kinda annoying, and prevents users from seeing how they used the command wrong if they did. 
- While not necessary, Id have no args also send the help embed. You can use [Deano-ActionHelp](https://github.com/sylo-digital/community-actions/tree/master/Snippets/Deano-ActionHelp) for that. 
- The time between an error message and it being deleted feels a bit to quick. I'd up it to around 10s or the 25(?)s that is native to Atlas
- Please use `{message.label}` along with the `{guild.prefx}` to dynamically update the command name in error and help messages.
- Not a required change, but your naming scheme for keys is very similar to tags and may cause confusing. Might I suggest changing from x.y to xY
- In the README you say to adjust `checks.bal` and `msg.reaction`. Both of which I can not find in your action. 
- In the README you misspelled embed, emnbed 
- What is the point of `role.manager`? It's not required to start a heist, and is only being used as a ping. Currently any user can start one of these.
- In the README you state its Customizable, that you can set an embed or a normal one, but there is no way to toggle between message or embed.
- What are these `{set;error.X}` at the start of the action? What do they do if they are just immediately being set to false and back.
- For the balance check, it is not looking for more than `pls bal` so one could check their own balance or some random users.

tl;dr it feels like I'm stating many of the same things Shylke did, and you haven't actually changed anything. And overall the action feels clunky.

**Comment by BrianWasTaken** [comment]
Alright I'm back.
